### PR TITLE
[RW-9990][risk=no]Converted how listPersistentDiskByProject handles admins.

### PR DESCRIPTION
**Pull request by evrii**
Updated listPersistentDiskByProject to not use a role when calling as an admin.

Ran unit DiskAdminControllerTest,DisksControllerTest, and DiskServiceTest locally and all passed. 
---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java [hunk 1]
```diff
@@ -439,14 +439,14 @@ public void updatePersistentDisk(String googleProject, String diskName, Integer
   @Override
   public List<LeonardoListPersistentDiskResponse> listPersistentDiskByProjectCreatedByCreator(
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:440]: I think we could go two ways with this.  If we do combine these, then I think we should also remove the indirection of calling the private `listPersistentDiskByProject()` and handle it all in this method.

OTOH: I like this separation.  We have a binary choice between "the normal flow" and "admin-flow" with a fixed set of parameters for each.  Even `includeDeleted` is a binary choice between admin and user-flow: it's always false for the latter, if you trace how it's called.  I want to remove the private `listPersistentDiskByProject()` in this case too: all of its logic is directed to making that same binary choice.

[the remark this answers is not in the document]
Comment (evrii) [reply api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java:440]: Hmmm, I agree with the second approach, but I have some questions about the second approach. 

I went with option two for now, but instead of getting rid of `listPersistentDiskByProject()`I changed its parameters to accept a `Provider<DisksApi>` and a String for a Leonardo. In that case, there is no admin logic in the function, it is just the logic that the two flows in common. In an attempt not to duplicate code, this common code is in a shared function. What do you think about that?

File: api/src/test/java/org/pmiops/workbench/disk/DiskServiceTest.java [hunk 2]
```diff
@@ -70,13 +71,12 @@ public void test_findByWorkspaceNamespace() {
             user,
             AppType.RSTUDIO);
     LeonardoListPersistentDiskResponse thirdLPDR =
-        createLeonardoListPersistentDiskResponse(
+        createLeonardoListRuntimePDResponse(
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/disk/DiskServiceTest.java:74]: leaving a note here: the reason for this change is to show that it works for the multiple-runtime-disks edge case relating to the bug of March/April

**Review by jmthibault79 — APPROVED** [review]
Tested locally: the admin version of the endpoint correctly shows Deleted and not-created-by-me disks, and the non-admin endpoint correctly does not show either of these.
